app/src/main/java/ch/epfl/sdp/appart/utils/PermissionRequest.java

```diff
@@ -90,6 +90,27 @@ public static void askForLocationPermission(Activity activity,
                         Manifest.permission.ACCESS_FINE_LOCATION});
     }
 
+    public static void askForActivityRecognitionPermission(Activity activity,
+                                                           Runnable permissionGranted,
+                                                           Runnable permissionRefused)  {
+        String[] permissions;
+
+        if (Build.VERSION.SDK_INT < 29) {
+            permissions = new String[]{
+                    Manifest.permission.READ_EXTERNAL_STORAGE,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE,
```

Comment:
I thought it was needed because it crashed the same way as the camera when the bug was discovered. But I tried it it works without demanding this. Good point!